### PR TITLE
[Merged by Bors] - feat(Algebra/Module): a presentation of an algebra induces a presentation of the module of differentials

### PR DESCRIPTION
**Pull request by joelriou**
---

- [x] depends on: #18408
- [x] depends on: #19504

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by joelriou** [comment]
Thanks very much @chrisflav for the reviews!

File: Mathlib/Algebra/Module/Presentation/Differentials.lean [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+  simp only [hom₁, Finsupp.linearCombination_single, one_smul, differentialsRelations,
+    Basis.repr_symm_apply, Extension.cotangentComplex_mk]
+  exact pres.cotangentSpaceBasis.repr.injective (by ext; simp)
+
+lemma comm₁₂ : pres.toExtension.cotangentComplex.comp (hom₁ pres) =
+    pres.cotangentSpaceBasis.repr.symm.comp (differentialsRelations pres).map := by
+  ext r
+  dsimp
+  rw [comm₁₂_single]
+  erw [Module.Relations.map_single]
```
Comment (chrisflav) [inline Mathlib/Algebra/Module/Presentation/Differentials.lean:120]:
```suggestion
  simp_rw [← differentialsRelations_G, ← differentialsRelations_R, Module.Relations.map_single]
```
The alternative to this is removing `@[simp]` from `differentialsRelations_{G,R}` and replace occurrences of `pres.vars` and `pres.rels` with `differentialsRelations.G` and `differentialsRelations.R` respectively.

Comment (joelriou) [reply Mathlib/Algebra/Module/Presentation/Differentials.lean]: Removing the `simps` attribute is not an option here, but I have found another way to remove the `erw` (this type of `erw` will be solved by the expected `dsimp%` elaborator).

**Comment by chrisflav** [comment]
Thanks!

maintainer merge

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**